npm/javy/tests/runner.js

```diff
@@ -0,0 +1,161 @@
+import { spawn } from "node:child_process";
+import * as stream from "node:stream";
+import { tmpdir } from "node:os";
+import { join } from "node:path";
+import { randomUUID } from "node:crypto";
+import {
+	ReadableStream,
+	TextDecoderStream,
+	TextEncoderStream,
+} from "node:stream/web";
+import { unlink } from "node:fs/promises";
+
+import { rollup } from "rollup";
+import swc from "rollup-plugin-swc";
+import nodeResolve from "@rollup/plugin-node-resolve";
+
+import * as tests from "./tests.js";
+
+const javyPath = new URL("../../../target/release/javy", import.meta.url)
+	.pathname;
+
+async function main() {
+	console.log("Running tests...");
+	const resultPromises = Object.entries(tests).map(([testName, testFunc]) =>
+		Promise.resolve(testFunc())
```

Comment:
Nit: Any reason not to use `await` on `testFunc()`?

Reply:
Instead of this, I could  do a `try/catch` + `await` (because I need to catch when `testFunc` throws). I found this more obvious, but I don't have a strong feeling.

Reply:
My personal style preference is to use `await` with a `try/catch` so the code can be read as if it were run sequentially. But it's not a super strong preference.